src/Libraries/RevitNodes/Elements/LinkElement.cs

```diff
@@ -0,0 +1,485 @@
+﻿using Autodesk.Revit.DB;
+using System;
+using Revit.Elements;
+using System.Collections.Generic;
+using System.Linq;
+using DynamoUnits;
+using Dynamo.Graph.Nodes;
+using View = Revit.Elements.Views.View;
+using System.Text;
+using System.Threading.Tasks;
+using Level = Revit.Elements.Level;
+using RevitServices.Persistence;
+using Autodesk.Revit.UI;
+using System.Windows;
+using Autodesk.DesignScript.Geometry;
+using Revit.GeometryConversion;
+using System.Windows.Media.Imaging;
+using Point = Autodesk.DesignScript.Geometry.Point;
+using Autodesk.Revit.UI;
+using System.Windows.Media.Media3D;
+using Autodesk.DesignScript.Runtime;
+using System.ComponentModel;
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// A Revit Link Instance
+    /// </summary>
+    public static class LinkElement
+    {
+
+        #region Helpers
+        private static List<RevitLinkInstance> GetLinkInstancesContainingLinkElement(Element linkElement)
+        {
+            Autodesk.Revit.DB.Element revitElement = linkElement.InternalElement;
+            // get the document of the linked element
+            Document linkDocToMatch = revitElement.Document;
+            Document currentDocument = Application.Document.Current.InternalDocument;
+            var allLinkInstances = new FilteredElementCollector(currentDocument)
+                .OfCategory(BuiltInCategory.OST_RvtLinks)
+                .WhereElementIsNotElementType()
+                .ToElements()
+                .Cast<RevitLinkInstance>()
+                .ToList();
+
+            // list to store all instances that match the linked element's document
+            List<RevitLinkInstance> matchingLinkInstances = new List<RevitLinkInstance>();
+            foreach (RevitLinkInstance linkInstance in allLinkInstances)
+            {
+                Document linkDoc = linkInstance.GetLinkDocument();
+                if (linkDoc.Equals(linkDocToMatch))
+                {
+                    matchingLinkInstances.Add(linkInstance);
+                }
+            }
+            return matchingLinkInstances;
+        }
+
+        // helper for zooming to clicked green Id
+        internal static void ZoomToLinkedElement(Element element)
+        {
+
+            UIDocument uiDoc = DocumentManager.Instance.CurrentUIDocument;
+            Autodesk.Revit.DB.View activeView = uiDoc.ActiveView;
+            // get active UI view to use
+            UIView uiview = uiDoc.GetOpenUIViews().FirstOrDefault<UIView>(uv => uv.ViewId.Equals(activeView.Id));
+            var location = Revit.Elements.LinkElement.GetLinkElementLocation(element);
```

Comment:
What if, by accident, the `linkElement` given is not actually a link element, but an element in the current Revit document?

Reply:
@Mikhinja the ZoomToLinkedElement only gets applied where the element's document does not equal the current document. 
```
                Document elementDoc = element.InternalElement.Document;
                Document currrentDoc = Revit.Application.Document.Current.InternalDocument;
                if (!(elementDoc.Equals(currrentDoc)) && (element.InternalElement.IsValidObject))
                {
                    Revit.Elements.LinkElement.ZoomToLinkedElement(element);
```
that otta do it, right? do you think there are any cases where this wouldn't work?

Reply:
@dariaivanciucova yes, that otta do it

Reply:
@dariaivanciucova Sorry for the pause in replies.
You don't seem to need the `location` here, should that line perhaps be removed?

Reply:
hey @Mikhinja done!